x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigVersion.java

```diff
@@ -153,12 +153,13 @@ private static void checkUniqueness(int id, String uniqueId) {
     // V_11 is used in ELSER v2 package configs
     public static final MlConfigVersion V_11 = registerMlConfigVersion(11_00_0_0_99, "79CB2950-57C7-11EE-AE5D-0800200C9A66");
     public static final MlConfigVersion V_12 = registerMlConfigVersion(12_00_0_0_99, "Trained model config prefix strings added");
+    public static final MlConfigVersion V_13 = registerMlConfigVersion(13_00_0_0_99, "Anomaly Detection reports actual memory usage");
 
     /**
      * Reference to the most recent Ml config version.
      * This should be the Ml config version with the highest id.
      */
-    public static final MlConfigVersion CURRENT = V_12;
+    public static final MlConfigVersion CURRENT = V_13;
```

Comment:
Is the config version change necessary? Model size stats are considered to be a result type not a config type and are stored in the .ml-anomalies-x indices.

ModelSizeStats has 2 parsers; a strict parser that errors if it does not recognise a field and a lenient parser that ignores unknown fields. The strict parser is used reading the output from autodetect and the lenient is used reading the documents from Elasticsearch. This way in a mixed version cluster if a upgraded node stores the stats doc with the `actual_memory_usage_bytes` field then when an old node reads the doc it won't error when it sees the unknown `actual_memory_usage_bytes` field. This mechanism works well for backwards compatibility and means that most of the time versioning is not required.